.github/workflows/sycl-linux-precommit.yml

```diff
@@ -60,7 +60,9 @@ jobs:
       - name: Determine Arc tests
         id: arc_tests
         run: |
-          if [ "${{ contains(needs.detect_changes.outputs.filters, 'drivers') }}" == "true" ]; then
+          if [ "${{ contains(needs.detect_changes.outputs.filters, 'devigccfg') }}" == "true" ]; then
+            echo 'arc_tests="Matrix/"' >> "$GITHUB_OUTPUT"
```

Comment:
I'd think we need to run all tests here as well.

Reply:
There are existing failures for non joint matrix tests.
eg: https://github.com/intel/llvm/actions/runs/8903576097/job/24452097484

We will only use igc dev driver for joint matrix for now, so we don't really need to run all e2e test,
only make sure joint matrix tests are clean is enough.

Running all tests would be an ideal solution, but to be honest, I don't think we have enough resource to triage and response to failures, so running all tests might effectively stop us from updating igc dev driver for joint matrix tests.

Reply:
What about ESIMD team's tests? + @sarnex

Reply:
TBH our tests take a long time to run. What is the use case of using the dev IGC driver? If I know that I can suggest if we need to run ESIMD tests with it.

Reply:
The dev IGC driver will be used in `joint matrix` pre-commit CI tests so that we can test some features that depends on newer IGC and get **integrated test** feedback earlier than before. 

I would personally recommend that we also enable the pre-commit CI for ESIMD, because ESIMD features tend to depends on corresponding IGC changes as well. But it is up to your team to decide.

The cost of enabling ESIMD test here is that we may need to response to potential failures in CI , it would be great if the failures are real regressions that we need to resolve, but there will definitely be noise due to IGC changes etc.

Reply:
Thanks and yeah I agree ESIMD is very sensitive to IGC versions as well. I think we should enable dev IGC driver testing in ESIMD.

@v-klochkov FYI